### PR TITLE
Add shoulda matchers o p

### PR DESCRIPTION
**Pull request by todtb**
Related #3155

### CODE REVIEW

**Comment by kueda** [comment]
Argh, sorry @todtb, I "shoulda" merged this before the ProjectInvitiation PR. Could you resolve the conflict here?

File: spec/models/observation_field_spec.rb [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
 describe ObservationField do
+  it { is_expected.to belong_to :user }
+  it { is_expected.to have_many(:observation_field_values).dependent :destroy }
+  it { is_expected.to have_many(:observations).through :observation_field_values }
+  it { is_expected.to have_many(:project_observation_fields).dependent :destroy }
+  it { is_expected.to have_many(:projects).through :project_observation_fields }
+  it { is_expected.to have_many(:comments).dependent :destroy }
+
+  it { is_expected.to validate_uniqueness_of(:name).case_insensitive }
+  # TODO: `validate_presence_of :name` contradicts `allow_blank` of length validation
```
Comment (todtb) [inline spec/models/observation_field_spec.rb:12]: I noticed in `observation_field.rb` the seeming contradiction:
```
validates_presence_of :name
validates_length_of :name, :maximum => 255, :allow_blank => true
```
As a user, my instinct is that `allow_blank` here is no longer applicable?

**Comment by todtb** [comment]
> Argh, sorry @todtb, I "shoulda" merged this before the ProjectInvitiation PR. Could you resolve the conflict here?

All set! I left a comment that you might have a real quick answer to, but not a blocker since the specs are marked pending.

File: spec/models/observation_field_spec.rb (hunk 1 above)
Comment (kueda) [reply spec/models/observation_field_spec.rb:12]: Yeah, an `ObservationField` needs a name, so I suspect we can get rid of `allow_blank`

File: spec/models/project_spec.rb [hunk 2]
```diff
@@ -1,6 +1,41 @@
 require File.dirname(__FILE__) + '/../spec_helper.rb'
 
 describe Project do
+  xit { is_expected.to belong_to :user }
+  xit { is_expected.to belong_to(:place).inverse_of :projects }
```
Comment (todtb) [inline spec/models/project_spec.rb:5]: With the 5.x switch to requiring `belongs_to` by default and then globally setting it to false, I've marked a few of these pending. I think the best solution would be to rewrite the association:

```ruby
belongs_to :user, optional: false, inverse_of: :projects
```
and remove the separate presence validation, but I'll leave those changes to another PR (hopefully after factories merges).

**Pull request merged**